### PR TITLE
feat(android_intent_plus): add ACTION_APN_SETTINGS intent support

### PR DESCRIPTION
**Pull request by hudovisk**
## Description

Adds support for [Settings.ACTION_APN_SETTINGS](https://developer.android.com/reference/android/provider/Settings#ACTION_APN_SETTINGS) intent

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by nohli** [comment]
Could you please also add it to
- docs
- example
- tests

**Comment by shilangyu** [comment]
Question: why are we introducing some aliases instead of just defining constants which store the exact strings? Would require no mapping nor custom logic

**Comment by miquelbeltran** [comment]
Hey! Due to changes in our contributor guidelines, all PRs should now:

- Have a title that follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) format. e.g. `feat(package_name)!: feature description`
- **DO NOT** modify the CHANGELOG.md or the version in the `pubspec.yaml`. This is going to be an automated process now.

We would ask you kindly to update the PR following these changes.

Thanks!

File: packages/android_intent_plus/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.2.0
+
+- Add `action_apn_settings` intent support
+
```
Comment (miquelbeltran) [inline packages/android_intent_plus/CHANGELOG.md:4]: Please revert this change as per the updated contributor guidelines (no need to update changelog or pubspec anymore)

File: packages/android_intent_plus/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: android_intent_plus
 description: Flutter plugin for launching Android Intents. Not supported on iOS.
-version: 3.1.2
+version: 3.2.0
```
Comment (miquelbeltran) [inline packages/android_intent_plus/pubspec.yaml:3]: Please revert (see previous comment)

**Comment by miquelbeltran** [comment]
> Question: why are we introducing some aliases instead of just defining constants which store the exact strings? Would require no mapping nor custom logic

I agree with this opinion. The implementation is a bit ridiculous, and we are not going to add ALL the [Android Settings](https://developer.android.com/reference/android/provider/Settings#ACTION_APN_SETTINGS).

You can accomplish the same goal just by passing `"android.settings.APN_SETTINGS"` as the action.

We will have to pass on this one, unfortunately. Thanks for the contribution @hudovisk nevertheless!
